iamclient/iamclient.go

```diff
@@ -78,6 +85,15 @@ type CertificateProvider interface {
 	GetCertSerial(certURL string) (serial string, err error)
 }
 
+// NodeInfoProvider provides information about the nodes.
```

Comment:
Interface should be defined in place where it used. Remove it.

Reply:
done